async-nats/src/connection.rs

```diff
@@ -671,6 +670,33 @@ mod read_op {
             })
         );
     }
+
+    #[tokio::test]
+    async fn unknown() {
+        let (stream, mut server) = io::duplex(128);
+        let mut connection = Connection {
+            stream: Box::new(stream),
+            buffer: BytesMut::new(),
+        };
+
+        server.write_all(b"ONE\r\n").await.unwrap();
```

Comment:
Would be nice to test HMSG cases here too? Especially those were the problematic ones.
And in general - all the proper proto cases? 😄

Reply:
Couple of cases, including empty HMSG cases are in the above block (message).